flake.nix

```diff
@@ -1,19 +1,21 @@
 {
   description = "Quickstart configurations for the Nvim LSP client";
 
-  inputs.nixpkgs.url = "github:NixOS/nixpkgs";
-  inputs.flake-utils.url = "github:numtide/flake-utils";
+  inputs = {
+    nixpkgs.url = "github:NixOS/nixpkgs";
+    flake-utils.url = "github:numtide/flake-utils";
+  };
 
   outputs = { self, nixpkgs, flake-utils }:
     flake-utils.lib.eachDefaultSystem (system:
       let pkgs = nixpkgs.legacyPackages.${system}; in
       rec {
-        devShell = pkgs.mkShell {
+        devShell = with pkgs; mkShell {
           buildInputs = [
-            pkgs.stylua
-            pkgs.luaPackages.luacheck
-            pkgs.luajitPackages.vusted
-            pkgs.selene
+            stylua
+            luaPackages.luacheck
+            luajitPackages.vusted
+            selene
```

Comment:
Instead of using `with`, which isn't a [very encouraged practice](https://nix.dev/guides/best-practices#with-scopes), use `builtin.attrValues` with `inherit`

For example
```nix
devShell = pkgs.mkShell {
  packages = builtin.attrValues {
    inherit (pkgs)
     stylua
     luaPackages.luacheck
     ...
     ;
  };
};
```

Also `packages` should be used over `buildInputs`, another nitpick of mine is that we can drop the `flake-utils` input entirely in favor of using `genAttrs` which provides a cleaner, readable Flake. Read this for more on that https://ayats.org/blog/no-flake-utils

Reply:
hmm I can see why it isn't encouraged but the points they mention are as follows:
> Static analysis can’t reason about the code, because it would have to actually evaluate this file to see which names are in scope.

> When more than one with used, it’s not clear anymore where the names are coming from.

> Scoping rules for with are not intuitive, see this [Nix issue for details](https://github.com/NixOS/nix/issues/490).

These don't apply for our specific use case here, simply using ```with pkgs``` is far more practical and intuitive here. That aside, I love that article regarding no-flake-utils, thank you for that :D

Reply:
❤️ no problem, I don't disagree with the `with` part but personally i just find the syntax nicer to read, but I do think flake utils should be removed

Reply:
yep, I'll do that as early as I can 👍